templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -4,28 +4,105 @@ contents:
-      dst_path="/etc/NetworkManager/system-connections"
-      if [ -d $src_path ]; then
+      local src_path="$NM_CONN_PATH"
+      local dst_path="$NM_CONN_UNDERLAY"
+      if [ "$src_path" = "$dst_path" ]; then
+        echo "No need to persist configuration files"
+        return
+      fi
+      if [ -d "$src_path" ]; then
         echo "$src_path exists"
-        # In RHEL7 files in /{etc,run}/NetworkManager/system-connections end without the suffix '.nmconnection', whereas in RHCOS they end with the suffix.
-        fileList=$(echo {br-ex,ovs-if-br-ex,ovs-port-br-ex,ovs-if-phys0,ovs-port-phys0} {br-ex,ovs-if-br-ex,ovs-port-br-ex,ovs-if-phys0,ovs-port-phys0}.nmconnection)
-        for file in ${fileList[*]}; do
-          if [ ! -f $dst_path/$file ] && [ -f $src_path/$file ]; then
-            cp $src_path/$file $dst_path/$file
+        local files=("${MANAGED_NM_CONN_FILES[@]}")
+        shopt -s nullglob
+        files+=("$src_path/*${MANAGED_NM_CONN_SUFFIX}.nmconnection")
+        shopt -u nullglob
+        for file in "${files[@]}"; do
+          file="$(basename $file)"
+          if [ -f "$src_path/$file" ]; then
+            if [ ! -f "$dst_path/$file" ]; then
+              echo "Persisting new configuration $file"
+              cp "$src_path/$file" "$dst_path/$file"
+            elif ! cmp --silent "$src_path/$file" "$dst_path/$file"; then
+              echo "Persisting updated configuration $file"
+              cp -f "$src_path/$file" "$dst_path/$file"
+            fi
           else
-            echo "Skipping $file since it exists in $dst_path"
+            echo "Skipping $file since its status is current"
+          fi
+        done
+      fi
+    }
+
+    # Used to remove files managed by configure-ovs
+    rm_nm_conn_files() {
+      local files=("${MANAGED_NM_CONN_FILES[@]}")
+      shopt -s nullglob
+      files+=("${NM_CONN_PATH}/*${MANAGED_NM_CONN_SUFFIX}.nmconnection")
+      shopt -u nullglob
+      for file in "${files[@]}"; do
+        file="$(basename $file)"
+        # Also remove files in underlay
+        for path in "${NM_CONN_PATH} ${NM_CONN_UNDERLAY}"; do
+          file_path="${path}/$file"
+          if [ -f "$file_path" ]; then
+            rm -f "$file_path"
+            echo "Removed nmconnection file $file_path"
           fi
         done
+      done
+    }
+
+    # Used to clone a slave connection by uuid, returns new uuid
+    clone_slave_connection() {
+      local uuid="$1"
+      local old_name
+      old_name="$(nmcli -g connection.id connection show uuid "$uuid")"
+      local new_name="${old_name}${MANAGED_NM_CONN_SUFFIX}"
+      if nmcli connection show id "${new_name}" &> /dev/null; then
+        echo "WARN: existing ovs slave ${new_name} connection profile file found, overwriting..."
+        nmcli connection delete id "${new_name}" &> /dev/null
       fi
+      nmcli connection clone $uuid "${new_name}" &> /dev/null
+      echo "$(nmcli -g connection.uuid connection show "${new_name}")"
```

Comment:
shellcheck suggests this `echo` is redundant maybe.

`note: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'. [SC2005]`

```
clone_slave_connection() {
  ...
  echo "$(nmcli -g connection.uuid connection show "${new_name}")"
}
    new_uuid=$(clone_slave_connection $conn_uuid)

```
we are using `$()` twice.  We could have `clone_slave_connection` return the raw `nmcli` command output and then the caller would capture with `$(clone_slave_connection $conn_uuid)`

```
clone_slave_connection() {
  ...
  nmcli -g connection.uuid connection show "${new_name}"
}

    new_uuid=$(clone_slave_connection $conn_uuid)

```